### PR TITLE
TEST: enable import tests for dataframes testing in `sklearnex.cluster.DBSCAN`

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
Draft PR just for checking how it works for now.
Probably make sense to check with #1832 

### TODO
* Create PR for this case:
  * https://github.com/intel/scikit-learn-intelex/blob/1b56189f238e5ee3d7976460ca3b65e36bf3be81/sklearnex/model_selection/tests/test_model_selection.py#L22

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by icfaust — APPROVED** [review]
looks good. Though the TSNE test is a bit weird with the dtype.

**Review by ethanglaser — APPROVED** [review]
